presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/NativeCassandraSession.java

```diff
@@ -221,24 +222,25 @@ public CassandraTable getTable(SchemaTableName schemaTableName)
         for (ColumnMetadata columnMeta : tableMeta.getPartitionKey()) {
             primaryKeySet.add(columnMeta.getName());
             boolean hidden = hiddenColumns.contains(columnMeta.getName());
-            CassandraColumnHandle columnHandle = buildColumnHandle(tableMeta, columnMeta, true, false, columnNames.indexOf(columnMeta.getName()), hidden);
+            CassandraColumnHandle columnHandle = buildColumnHandle(tableMeta, columnMeta, true, false, columnNames.indexOf(columnMeta.getName()), hidden)
+                    .orElseThrow(() -> new PrestoException(NOT_SUPPORTED, "Unsupported primary key type: " + columnMeta.getType().getName()));
```

Comment:
"primary key" -> "partition key" ?
(i am not sure)

Reply:
It depends on the column numbers. When the primary key is only one, it's called primary key. When more than one primary keys, left key is called partition key and right key is called clustering key.
- `PRIMARY KEY(key_primary)`
- `PRIMARY KEY(key_partition_1, key_clustering_1)`
- `PRIMARY KEY((k_partition_1, k_partition_2), k_clustering_1, k_clustering_2)`

Reply:
Thanks.
This loop is over `tableMeta.getPartitionKey()`. Clustering keys (`tableMeta.getClusteringColumns()`) are processed in the loop below.

(btw, did you run test what happens if one of the clustering keys is not supported?)

Reply:
Yes, while we can select, insert fails with below message for missing clustering keys.

client log
```
presto> insert into keyspace_6.table_6 select 1;

Query 20190415_133608_00009_m2ru3 failed: Some clustering keys are missing: unsupported_clustering_key
```

server log
```
com.datastax.driver.core.exceptions.InvalidQueryException: Some clustering keys are missing: unsupported_clustering_key
	at com.datastax.driver.core.exceptions.InvalidQueryException.copy(InvalidQueryException.java:50)
	at com.datastax.driver.core.DriverThrowables.propagateCause(DriverThrowables.java:37)
	at com.datastax.driver.core.AbstractSession.prepare(AbstractSession.java:117)
	at io.prestosql.plugin.cassandra.NativeCassandraSession.lambda$prepare$10(NativeCassandraSession.java:449)
	at io.prestosql.plugin.cassandra.NativeCassandraSession.executeWithSession(NativeCassandraSession.java:553)
	at io.prestosql.plugin.cassandra.NativeCassandraSession.prepare(NativeCassandraSession.java:449)
	at io.prestosql.plugin.cassandra.CassandraPageSink.<init>(CassandraPageSink.java:104)
	at io.prestosql.plugin.cassandra.CassandraPageSinkProvider.createPageSink(CassandraPageSinkProvider.java:72)
	at io.prestosql.split.PageSinkManager.createPageSink(PageSinkManager.java:61)
	at io.prestosql.operator.TableWriterOperator$TableWriterOperatorFactory.createPageSink(TableWriterOperator.java:114)
	at io.prestosql.operator.TableWriterOperator$TableWriterOperatorFactory.createOperator(TableWriterOperator.java:105)
	at io.prestosql.operator.DriverFactory.createDriver(DriverFactory.java:114)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunnerFactory.createDriver(SqlTaskExecution.java:941)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1069)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_null__testversion____20190415_132251_1.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.datastax.driver.core.exceptions.InvalidQueryException: Some clustering keys are missing: unsupported_clustering_key
	at com.datastax.driver.core.Responses$Error.asException(Responses.java:136)
	at com.datastax.driver.core.SessionManager$4.apply(SessionManager.java:220)
	at com.datastax.driver.core.SessionManager$4.apply(SessionManager.java:196)
	at com.datastax.driver.$internal.com.google.common.util.concurrent.Futures$ChainingListenableFuture.run(Futures.java:861)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:145)
	... 1 more
```

Reply:
@ebyhr thanks for checking this.

Considering the above, I think the message should be:
```
``

Reply:
@ebyhr thanks for checking this.

Considering the above, I think the message should be:
```suggestion
                    .orElseThrow(() -> new PrestoException(NOT_SUPPORTED, "Unsupported partition key type: " + columnMeta.getType().getName()));
```

Reply:
Updated the message.